src/components/SplitField/SplitField.jsx

```diff
@@ -49,13 +56,17 @@ export const SplitField = forwardRef((props, ref) => {
         </InputGroup>
       )}
 
-      <Card className="split-field__amount">$0</Card>
+      <Card className="split-field__amount">
+        $
+        {props.splitType === EQUALLY &&
+          formatCurrency(props.totalExpense / props.splits.length)}
+        {props.splitType === EXACT_AMOUNTS && input}
```

Comment:
This gets the job done of displaying the split amount here, but this way does not allow the split amount to be accessed by other components, such as when we need the sum of all split amounts to display the running total and how much is left from the total expense to be split. 

What we probably want to do is have the value of each split stored higher up and a good place to be that would be in the split object that exists inside of `splits`. This also would allow us to handle that situation when the total split expense cannot be split equally by the same amount.

```jsx
// App.js

const createNewSplit = () => ({
  id: generateUuid(),
  name: '',
  value: 0, // add new value attribute
  nodeRef: createRef(null),
})

const initialSplits = [createNewSplit()];

function App() {
  const [selectedSplitType, setSelectedSplitType] = useState(EQUALLY);
  const [splits, setSplits] = useState(initialSplits);
  const [totalExpense, setTotalExpense] = useState("");

  const handleSplitTypeSelect = (type) => {
    setSelectedSplitType(type);
    
    if (type === EQUALLY) {
      // Since a penny is the lowest denomination we can deal with,
      // it's easier to handle this in cents
      const totalExpenseInCents = totalExpense * 100;
      
      // The count of splits that will need an extra penny added
      const splitsWithExtraPennyCount = totalExpenseInCents % splits.length;
      
      // Lowest amount we can split equally
      // Notice I added the + operator in front. This is shorthand for turning a 
      // string into a number, since toFixed(2) returns a string. We want to turn 
      // this back into a number so that we can perform math operations to this
      // value correctly.
      const equalSplitAmount = +(totalExpense / splits.length).toFixed(2)
      
      const newSplits = splits.map((split, index) => {
        if (index + 1 <== splitsWithExtraPennyCount) {
          return { ...split, value: equalSplitAmount + 0.01 };
        }
        
        return { ...split, value: equalSplitAmount };
      })
      
      setSplits(newSplits);
    }
  };
  
  return (
    <div>
      <ActionBar onSplitTypeSelect={handleSplitTypeSelect} />
      <Splits selectedSplitType={selectedSplitType} splits={splits} />
    </div>
  )
}
```

Since I've exposed `onSplitTypeSelect` as an event callback prop, I'm able to add some additional logic that I want handled when this event happens (this is one of those benefits I've mentioned in the comment for `App.js`). 

Here, if the split type is `EQUALLY`, then we can change the value of all splits to be equal or close to it. We can determine how to handle those "missing pennies" by using the [remainder operator (`%`)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder). By first multiplying the `totalExpense` by 100, we can get the total expense amount in cents. Then, with the remainder operator, `totalExpenseInCents % splits.length` gives us how many cents are remainder if we were to divide `totalExpenseInCents` by the amount of splits. For example, given there are 3 splits and a total expense of 1000:

```js
const amountOfSplits = 3
const totalExpense = 1000
const totalExpenseInCents = totalExpense * 100 // 10000

// There is 1 extra penny, so 1 split will get an extra penny
const splitsWithExtraPennyCount = totalExpenseInCents % amountOfSplits // 1

// All other splits will get 333.33
const equalSplitAmount = +(totalExpense / amountOfSplits).toFixed(2) // 333.33

// For 3 splits with a total expense of 1000, one split will 
// get an extra penny while two will get 333.33
// 333.33 + 333.33 + 333.34 = 1000
```

Now you can pass the `split.value` as an argument to the prop `value` for `SplitField` (you will need to add this)
```jsx
// Splits.jsx

export const Splits (props) => {
  return (
    <div>
      {props.splits.map((split, index) => (
        <SplitField
          ref={split.nodeRef}
          value={split.value} // pass the value through as a prop
          position={index + 1}
          splitType={props.selectedSplitType}
          onRemove={handleRemoveFor(split.id)}
        />
      ))}
    </div>
  )
}
```

```jsx
// SplitField.jsx
export const SplitField = (props) => {
  return (
    <Card>${formatCurrency(props.value)}</Card>
  )
}
```

I will let you try to figure out how to handle changing the splits value when the split type is `EXACT_AMOUNTS`, but let me know if you need any help or any clarification for this 👍🏼

Reply:
Just a disclaimer, this doesn't handle changing the split amounts when splits are added or removed, so I would like to see how you might handle that 😉